lib/SILGen/SILGenApply.cpp

```diff
@@ -3340,12 +3341,41 @@ void DelayedArgument::emitDefaultArgument(SILGenFunction &SGF,
                                           const DefaultArgumentStorage &info,
                                           SmallVectorImpl<ManagedValue> &args,
                                           size_t &argIndex) {
-  auto value = SGF.emitApplyOfDefaultArgGenerator(info.loc,
-                                                  info.defaultArgsOwner,
-                                                  info.destIndex,
-                                                  info.resultType,
-                                                  info.origResultType);
-  
+  RValue value;  
+  bool isStoredPropertyDefaultArg = false;
+
+  // Check if this is a synthesized memberwise constructor for stored property
+  // default values
+  if (auto ctor = dyn_cast<ConstructorDecl>(info.defaultArgsOwner.getDecl())) {
+    if (ctor->isMemberwiseInitializer() && ctor->isImplicit()) {
+      auto param = ctor->getParameters()->get(info.destIndex);
```

Comment:
This feels a bit fragile. Is there a way to get the ParamDecl from the DefaultArgumentStorage instead?